src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditCategory.java

```diff
@@ -0,0 +1,34 @@
+package com.amazon.opendistroforelasticsearch.security.auditlog.impl;
+
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public enum AuditCategory {
+    BAD_HEADERS,
+    FAILED_LOGIN,
+    MISSING_PRIVILEGES,
+    GRANTED_PRIVILEGES,
+    OPENDISTRO_SECURITY_INDEX_ATTEMPT,
+    SSL_EXCEPTION,
+    AUTHENTICATED,
+    COMPLIANCE_DOC_READ,
+    COMPLIANCE_DOC_WRITE,
+    COMPLIANCE_EXTERNAL_CONFIG,
+    COMPLIANCE_INTERNAL_CONFIG_READ,
+    COMPLIANCE_INTERNAL_CONFIG_WRITE;
+
+    public static EnumSet<AuditCategory> parse(final Collection<String> categories) {
+        EnumSet<AuditCategory> set = EnumSet.noneOf(AuditCategory.class);
+        if (categories == null)
```

Comment:
Is it necessary to allow `null`? Should it check for an empty collection and throw `NullPointerException`?

Reply:
I don't think so. As far as this parse is concerned, absence of values in input (whether null || empty) should mean no output categories.

Reply:
While you may handle `null` the same way as an empty collection, the `parse()` function itself follows "prefer empty collection over null" and converts `null` to an empty set.

Reply:
I didnt quite follow you. 
If we have an input list of nulls, since we are filtering non-nulls in the stream we would return empty set. 
This is what I meant by absence of values and we are return an empty set. [null, null...] = null => return empty set.

Reply:
What if the collection `categories` itself is `null`? When I referred to "check for an empty collection" I meant instead of `if (categories == null)`, check `if (categories.isEmpty())`.

You brought a good question, how `null` should be handled inside the collection. IMO, that is a programming mistake to pass `null` inside the categories collection.

Reply:
Are you saying to short-circuit `if (categories == null || categories.isEmpty())`  ?
I mean to say, absence of values whether categories is null or empty or list of only null strings should all mean the same thing -> empty category set.

Reply:
@vrozov If you could tell exactly the change you are looking for, i could can do that and move on. I don't want to be blocked with such issues.

Reply:
My suggestion is not to allow `null` at all, whether the collection itself is `null` or it contains `null`. Change `if (categores == null)` to `if (categories.isEmpty())` and remove the first `filter`.